shub/settings.py

```diff
@@ -0,0 +1,537 @@
+    "django_extensions",
+    "rest_framework",
+    "rest_framework.authtoken",
+    "rest_framework_swagger",
+    "taggit",
+]
+
+
+INSTALLED_APPS += THIRD_PARTY_APPS
+
+
+# Authentication
+
+# Python-social-auth
+
+AUTHENTICATION_BACKENDS = (
+    "django.contrib.auth.backends.ModelBackend",
+    "social_core.backends.username.UsernameAuth",
+    "social_core.backends.open_id.OpenIdAuth",
+    "social_core.backends.saml.SAMLAuth",
+    "social_core.backends.google.GoogleOAuth2",
+    "social_core.backends.twitter.TwitterOAuth",
+    "social_core.backends.facebook.FacebookOAuth2",
+    "shub.apps.users.views.auth.ShubGithubOAuth2",
+    # "social_core.backends.github.GithubOAuth2",
+    "social_core.backends.github_enterprise.GithubEnterpriseOAuth2",
+    "social_core.backends.gitlab.GitLabOAuth2",
+    "social_core.backends.bitbucket.BitbucketOAuth2",
+)
+
+
+SOCIAL_AUTH_PIPELINE = (
+    "social_core.pipeline.social_auth.social_details",
+    "social_core.pipeline.social_auth.social_uid",
+    "social_core.pipeline.social_auth.auth_allowed",
+    "shub.apps.users.views.social_user",
+    "shub.apps.users.views.redirect_if_no_refresh_token",
+    "social_core.pipeline.user.get_username",
+    "social_core.pipeline.social_auth.associate_by_email",  # <--- must share same email
+    "social_core.pipeline.user.create_user",
+    "social_core.pipeline.social_auth.associate_user",
+    "social_core.pipeline.social_auth.load_extra_data",
+    "social_core.pipeline.user.user_details",
+)
+
+
+# If these need to be exposed via envar, we can do.
+# http://psa.matiasaguirre.net/docs/configuration/settings.html#urls-options
+# SOCIAL_AUTH_REDIRECT_IS_HTTPS = True
+# SOCIAL_AUTH_USER_MODEL = 'django.contrib.auth.models.User'
+# SOCIAL_AUTH_STORAGE = 'social.apps.django_app.me.models.DjangoStorage'
+# SOCIAL_AUTH_LOGIN_REDIRECT_URL = '/'
+# SOCIAL_AUTH_LOGIN_REDIRECT_URL = '/logged-in/'
+# SOCIAL_AUTH_LOGIN_ERROR_URL = '/login-error/'
+# SOCIAL_AUTH_LOGIN_URL = '/login-url/'
+# SOCIAL_AUTH_NEW_USER_REDIRECT_URL = '/new-users-redirect-url/'
+# SOCIAL_AUTH_LOGIN_REDIRECT_URL
+# SOCIAL_AUTH_NEW_ASSOCIATION_REDIRECT_URL = '/new-association-redirect-url/'
+# SOCIAL_AUTH_DISCONNECT_REDIRECT_URL = '/account-disconnected-redirect-url/'
+# SOCIAL_AUTH_INACTIVE_USER_URL = '/inactive-user/'
+
+
+# AUTHENTICATION
+
+DOMAIN_NAKED = cfg.DOMAIN_NAME_HTTP.replace("http://", "")
+
+envar_admins = get_sregistry_envar_list("SREGISTRY_ADMINS")
+ADMINS = (("vsochat", "@vsoch"),)
+if envar_admins:
+    ADMINS = tuple(envar_admins)
+MANAGERS = ADMINS
+
+
+# DATABASE
+
+# https://docs.djangoproject.com/en/1.9/ref/settings/#databases
+
+DATABASES = {
+    "default": {
+        "ENGINE": cfg.SREGISTRY_DATABASE_ENGINE,
+        "NAME": cfg.SREGISTRY_DATABASE_NAME,
+        "USER": cfg.SREGISTRY_DATABASE_USER,
+        "HOST": cfg.SREGISTRY_DATABASE_HOST,
+        "PORT": cfg.SREGISTRY_DATABASE_PORT,
+    }
+}
+
+# STORAGE
+
+MINIO_ROOT_USER = os.environ.get("MINIO_ROOT_USER")
```

Comment:
While we're here, why not allow this to be set via the config as well? To keep backwards compatibility we could do:
```
MINIO_ROOT_USER = os.environ.get("MINIO_ROOT_USER", cfg.MINIO_ROOT_USER)
```
and make the default for `MINIO_ROOT_USER` in the `STRING_DEFAULTS` to be `None`.

Reply:
We could definitely do that, although for the actual Minio server it is expecting `MINIO_ROOT_USER` explicitly. I think the use case that you have in mind is having deployed your Minio separately (and so you could use `SREGISTRY_MINIO_ROOT_USER` without an issue? I'm definitely happy to add that!

Reply:
I also realize that I need to do a check if a value is set before setting to locals, e.g., for the case that the cfg MINIO_ROOT_USER (from `SREGISTRY_MINIO_ROOT_USER) is None, we don't want it to override `MINIO_ROOT_USER`

Reply:
Correct - I need to decouple the two. Eventually I need this to work against a genuine S3 bucket rather than a minio server, but I haven't yet gotten to testing that / making changes to support it.

Reply:
Oh nice! That would be amazing to have (and based on the shared protocol hopefully not too hard to support). I can definitely help when the time comes.